pkg/collector/pv_collector.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2021 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/klog"
+
+	"github.com/openebs/lvm-localpv/pkg/lvm"
+)
+
+/*
+pvCollector collects all the necessary metrics related to physical volume
+*/
+type pvCollector struct {
+	pvSizeMetric         *prometheus.Desc
+	pvFreeMetric         *prometheus.Desc
+	pvUsedMetric         *prometheus.Desc
+	pvDeviceSizeMetric   *prometheus.Desc
+	pvMetadataSizeMetric *prometheus.Desc
+	pvMetadataFreeMetric *prometheus.Desc
+}
+
+func NewPvCollector() prometheus.Collector {
+	return &pvCollector{
+		pvSizeMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "pv", "total_size_bytes"),
+			"LVM PV total size in bytes",
+			[]string{"name", "allocatable", "vg", "missing", "in_use"}, nil,
+		),
+		pvFreeMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "pv", "free_size_bytes"),
+			"LVM PV free size in bytes",
+			[]string{"name", "allocatable", "vg", "missing", "in_use"}, nil,
+		),
+		pvUsedMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "pv", "used_size_bytes"),
+			"LVM PV used size in bytes",
+			[]string{"name", "allocatable", "vg", "missing", "in_use"}, nil,
+		),
+		pvDeviceSizeMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "pv", "device_size_bytes"),
+			"LVM PV underlying device size in bytes",
```

Comment:
what does underlying device size mean here?

Reply:
The actual hard drive